### PR TITLE
Motherlode Mine - Gem overlay

### PR DESCRIPTION
**Pull request by ypperlig**
When mining with an alumet of glory the chance of receiving uncut gems increases significantly.

This introduces an overlay to the Motherlode Mine Plugin which shows the numer of gems found during the current mining session.

Screenshot:
image

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodePlugin.java [hunk 1]
```diff
@@ -83,6 +83,10 @@
 			ItemID.MITHRIL_ORE, ItemID.GOLD_ORE, ItemID.COAL, ItemID.GOLDEN_NUGGET);
 	private static final Set<Integer> ROCK_OBSTACLES = ImmutableSet.of(ROCKFALL, ROCKFALL_26680);
 
+	@Getter(AccessLevel.PACKAGE)
+	private static final Set<Integer> MOTHERLODE_GEM_TYPES = ImmutableSet.of(ItemID.UNCUT_DIAMOND, ItemID.UNCUT_RUBY,
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodePlugin.java]: This field is unnecessary .. just put a default case in the switch

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeSession.java [hunk 2]
```diff
@@ -39,6 +42,47 @@
...
+	@Getter(AccessLevel.PACKAGE)
+	private int rubiesFound;
+
+	@Getter(AccessLevel.PACKAGE)
+	private int emeraldsFound;
+
+	@Getter(AccessLevel.PACKAGE)
+	private int sapphiresFound;
+
+	public void incrementGemFound(int GemID)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeSession.java]: Use camel case for variable names

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeGemOverlay.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+		Duration statTimeout = Duration.ofMinutes(config.statTimeout());
+		Duration sinceCut = Duration.between(session.getLastGemFound(), Instant.now());
+
+		if (sinceCut.compareTo(statTimeout) >= 0)
+		{
+			return null;
+		}
+
+		if (!plugin.isInMlm() || !config.showGemsFound())
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeGemOverlay.java]: These two checks should go up on line 32

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeGemOverlay.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
+package net.runelite.client.plugins.motherlode;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeGemOverlay.java]: Add license header to new files

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java [hunk 5]
```diff
@@ -95,4 +95,14 @@ default boolean showMiningState()
 	{
 		return true;
 	}
+
+	@ConfigItem(
+			keyName = "showGemsFound",
+			name = "Show gems found",
+			description = "Shows gems found during current mining session"
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java]: Fix indenting

**Comment by SoyChai** [comment]
We mentioned this on discord, but isn't this change applicable to gem mining in general, not just the Motherlode Mine?

**Comment by ypperlig** [comment]
I agree it should probably be in a mining specific plugin. I can create a mining specific plugin instead and PR that.

If I create a mining specific plugin, should I merge the Motherlode Mine plugin with it or leave it as is and exclude Motherlode Mine from the Mining plugin?

**Comment by SoyChai** [comment]
IMHO I think it's worth keeping them in the same plugin, and just renaming it to `Mining`, unless anyone else has an objection. I don't think the Motherlode's mechanics are unique to deserve their own plugin, and it's not even listed on the Minigames list.

**Pull request merged**